code/modules/mob/living/carbon/carbon.dm

```diff
@@ -536,7 +544,15 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
 		for(var/image/current_image in pipes_shown)
 			client.images -= current_image
 		client.eye = src
-
+	if(mind)
+		var/obj/effect/proc_holder/spell/reaper_lighter/S = locate(/obj/effect/proc_holder/spell/reaper_lighter) in mind.spell_list
```

Comment:
I feel this should be handled with a signal instead. A signal that is being sent here that the mob exited a vent. And the effect, or spell, listening to this signal.